pkgs/by-name/ba/balatro/package.nix

```diff
@@ -0,0 +1,79 @@
+{
+  stdenv,
+  lib,
+  pkgs,
+  makeDesktopItem,
+  requireFile,
+  withMods ? true,
+}:
+let
+  version = "1.0.1g";
+  balatroExe = requireFile {
+    name = "Balatro-${version}.exe";
+    url = "https://store.steampowered.com/app/2379780/Balatro/";
+    # Use `nix hash file --sri --type sha256` to get the correct hash
+    hash = "sha256-X7t/T3lkLnDzewlmkxHjdePH/hlwoachuWfoUC4ntd0=";
+  };
+in
+stdenv.mkDerivation {
+  pname = "balatro";
+  inherit version;
+  nativeBuildInputs = with pkgs; [
+    p7zip
+    copyDesktopItems
+  ];
+  buildInputs = with pkgs; [ love ] ++ lib.optional withMods pkgs.lovely-injector;
+  src = ./.;
+  desktopItems = [
+    (makeDesktopItem {
+      name = "balatro";
+      desktopName = "Balatro";
+      exec = "balatro";
+      keywords = [ "Game" ];
+      categories = [ "Game" ];
+      icon = "balatro";
+    })
+  ];
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p $out/share/{applications,icons}
+
+    # "Patch" game and extract icon
+    tmpdir=$(mktemp -d)
```

Comment:
We're already in a sandboxed build dir, I don't see why we need to create another temp dir. Creating another dir in the CWD would be fine.

Reply:
Trying to put a static path intsead of `mktemp` (E.G. `BalatroExtracted`) results in a non-working game. That probably has to do with the re-zipping process which results in a wrong directory structure, resulting in love not being able to find the game.